src/controller.cpp

```diff
@@ -602,12 +605,16 @@ bool Controller::processNextStep() {
   }
 
   if (nextStep == BackendFailure) {
+    emit vpn->recordGleanEvent(GleanSample::backendFailureError);
+
     emit readyToBackendFailure();
     return true;
   }
 
   if (nextStep == ServerUnavailable) {
     logger.info() << "Server Unavailable - Ping succeeded: " << m_ping_received;
+    emit vpn->recordGleanEvent(GleanSample::serverUnavailableError);
```

Comment:
- the `readyToServerUnavailable` is emitted in a few places
- Can we record the event in the MozillaVPN class? Or even better in `telementry.cpp` ?